### DISCUSSION TITLE
How to detect a render trigger?

### DISCUSSION THREAD

**Discussion by vadymshymko** [vercel/next.js#36404] [Help] [answered]
Is it possible to detect that the page rendering process was started as part of the ISR process and not as part of the rendering at runtime in the browser

Any help is appreciated

**Comment by icyJoseph** [comment, marked as answer]
When you say ISR, I assume the page has `getStaticProps` implemented, which by definition runs only server side. So you could send a ping to another service, or write to the file system, whenever the function runs. I think the only issue would be that at build time you'd also run this logic, but you could aggregate more data about the process which build the page, or make sure that the `process.argv` list includes `start`. 

For example, in this [page](https://github.com/icyJoseph/next-js-react-gbg/blob/main/demo-app/src/pages/pokemon/%5Bid%5D.tsx), we could add a `console.log` statement:

```ts
export const getStaticProps: GetStaticProps<{ pokemon: Pokemon }> = async (
  ctx
) => {
  const id = ctx.params?.id;

  const idx = Number(id);
  
  // `argv` would include `start` when the server is running with `next start`
  console.log({ id, cwd: process.cwd(), argv: process.argv }); 

  if (Number.isNaN(idx)) return { notFound: true };

  try {
    const pokemon = await fetchPokemon(idx);

    return {
      props: { pokemon },
      revalidate: 900,
    };
  } catch (e) {
    console.log(e);
    return { notFound: true, revalidate: 500 };
  }
};
```

And instead of just logging you could send this data to an aggregation system.

**Reply by vadymshymko** [reply]
Thanks a lot!
